server/service/service.go

```diff
@@ -11,14 +11,17 @@ import (
 	"github.com/fleetdm/fleet/v4/server/config"
 	"github.com/fleetdm/fleet/v4/server/fleet"
 	"github.com/fleetdm/fleet/v4/server/logging"
+	"github.com/fleetdm/fleet/v4/server/service/async"
 	"github.com/fleetdm/fleet/v4/server/sso"
 	kitlog "github.com/go-kit/kit/log"
 	"github.com/pkg/errors"
 )
 
 // Service is the struct implementing fleet.Service. Create a new one with NewService.
 type Service struct {
-	ds             fleet.Datastore
+	ds fleet.Datastore
+	// TODO: make that an interface, create a generated mock for tests?
```

Comment:
I don't think it's worth making an interface in this case, as the Task depends on both fleet.Datastore and fleet.RedisPool, so both of those can be easily mocked.